syntax = "proto3";

package v1;

option go_package = "neuro_search/gateway/pkg/api/v1";

service RagService {
  rpc GetAnswer (ChatRequest) returns (ChatResponse);
}

message ChatRequest {
  string message = 1;
  repeated MessageHistory history = 2;
  string session_id = 3;  // Optional: existing session ID
}

message MessageHistory {
  string role = 1;
  string content = 2;
}

message ChatResponse {
  string answer = 1;
  repeated Source sources = 2;
  string session_id = 3;  // Session ID for this conversation
}

message Source {
  string doc_name = 1;
  int32 page = 2;
  float score = 3;
}
